lib/brain.js

```diff
@@ -660,12 +660,26 @@ Brain.prototype.timeoutToScannerCancel = function timeoutToScannerCancel (stateM
   }, _.get('scanner.timeout', this.rootConfig))
 }
 
+Brain.prototype.idCardStillsCallback = function idCardStillsCallback (photo) {
+  idCardData.addIdDataPhoto(photo.toString('base64'))
+}
+
 Brain.prototype.scanPDF = function scanPDF () {
+  const customer = this.customer
   this.scanBayLightOn()
   this.scanner.scanPDF417((err, result) => {
     this.scanBayLightOff()
     this.startDisabled = false
 
+    if (err || !result) {
+      const photos = idCardData.getIdDataPhotos()
+      this.trader.updateIdCardPhotos(customer.id, { photos })
+        .then(() => console.log('Successfully saved ', photos.length, ' id data photos'))
+        .catch(err => {
+          console.log('Error saving id card photos', err)
+        })
+    }
```

Comment:
Don't we have that problem here were we pass through this code multiple times during a single id scan try?